File: app/src/main/java/com/example/marta/pin2/viewmodels/PFPinCodeViewModel.java
package com.example.marta.pin2.viewmodels;


import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.marta.pin2.security.PFResult;
import com.example.marta.pin2.security.PFSecurityManager;
import com.example.marta.pin2.security.callbacks.PFPinCodeHelperCallback;
import com.example.marta.pin2.security.livedata.PFLiveData;


public class PFPinCodeViewModel extends ViewModel {

    public LiveData<PFResult<String>> encodePin(Context context, String pin){
        final PFLiveData<PFResult<String>> liveData= new PFLiveData<>();
        PFSecurityManager.getInstance().getPinCodeHelper().encodePin(context, pin,
                new PFPinCodeHelperCallback<String>() {
                     @Override
                    public void onResult(PFResult result) {
                        liveData.setData(result);
                    }
                }
        );
        return  liveData;
    }
    public LiveData<PFResult<Boolean>> checkPin(Context context, String encodedPin, String pin) {
        final PFLiveData<PFResult<Boolean>> liveData = new PFLiveData<>();
        PFSecurityManager.getInstance().getPinCodeHelper().checkPin(
                context,
                encodedPin,
                pin, new PFPinCodeHelperCallback<Boolean>() {
                    @Override
                    public void onResult(PFResult result) {
                        liveData.setData(result);
                    }
                }

        );
        return liveData;
    }

    public LiveData<PFResult<Boolean>> delete() {
        final PFLiveData<PFResult<Boolean>> liveData = new PFLiveData<>();
        PFSecurityManager.getInstance().getPinCodeHelper().delete(
                new PFPinCodeHelperCallback<Boolean>() {
                    @Override
                    public void onResult(PFResult result) {
                        liveData.setData(result);
                    }
                }
        );
        return liveData;
    }

    public LiveData<PFResult<Boolean>> isPinCodeEncryptionKeyExist() {
        final PFLiveData<PFResult<Boolean>> liveData = new PFLiveData<>();
        PFSecurityManager.getInstance().getPinCodeHelper().isPinCodeEncryptionKeyExist(
                new PFPinCodeHelperCallback<Boolean>() {
                    @Override
                    public void onResult(PFResult result) {
                        liveData.setData(result);
                    }
                }

        );
        return liveData;
    }

}

File: app/src/main/java/com/example/marta/ui/pin/viewmodels/PFPinCodeViewModel.kt
//package com.example.marta.ui.pin.viewmodels
//
//import android.content.Context
//import androidx.lifecycle.LiveData
//import androidx.lifecycle.ViewModel
//import com.example.marta.ui.pin.security.PFResult
//import com.example.marta.ui.pin.security.PFSecurityManager
//import com.example.marta.ui.pin.security.calbacs.PFPinCodeHelperCallback
//import com.example.marta.ui.pin.security.liveData.PFLiveData
//
//
//class PFPinCodeViewModel : ViewModel() {
//    fun encodePin(
//        context: Context?,
//        pin: String?
//    ): LiveData<PFResult<String>> {
//        val liveData: PFLiveData<PFResult<String>> = PFLiveData()
//        PFSecurityManager.instance.pinCodeHelper.encodePin(
//            context,
//            pin,
//            object : PFPinCodeHelperCallback<String> {
//                fun onResult(result: PFResult<Boolean>) {
//                    liveData.setData(result)
//                }
//
//                override fun onResult(result: PFResult<String>) {
//                    TODO("Not yet implemented")
//                }
//            }
//        )
//        return liveData
//    }
//
//    fun checkPin(
//        context: Context?,
//        encodedPin: String?,
//        pin: String?
//    ): LiveData<PFResult<Boolean>> {
//        val liveData: PFLiveData<PFResult<Boolean>> = PFLiveData()
//        PFSecurityManager.instance.pinCodeHelper.checkPin(
//            context,
//            encodedPin,
//            pin,
//            object : PFPinCodeHelperCallback<Boolean?> {
//                fun onResult(result: PFResult<Boolean?>?) {
//                    liveData.setData(result)
//                }
//            }
//        )
//        return liveData
//    }
//
//    fun delete(): LiveData<PFResult<Boolean>> {
//        val liveData: PFLiveData<PFResult<Boolean>> = PFLiveData()
//        PFSecurityManager.instance.pinCodeHelper.delete(
//            object : PFPinCodeHelperCallback<Boolean?>() {
//                fun onResult(result: PFResult<Boolean?>?) {
//                    liveData.setData(result)
//                }
//            }
//        )
//        return liveData
//    }
//
//    val isPinCodeEncryptionKeyExist: LiveData<Any>
//        get() {
//            val liveData: PFLiveData<PFResult<Boolean>> = PFLiveData()
//            PFSecurityManager.instance.pinCodeHelper.isPinCodeEncryptionKeyExist(
//                object : PFPinCodeHelperCallback<Boolean?>() {
//                    fun onResult(result: PFResult<Boolean?>?) {
//                        liveData.setData(result)
//                    }
//                }
//            )
//            return liveData
//        }
//}